base_layer/p2p/examples/pingpong.rs

```diff
@@ -176,39 +173,38 @@ mod pingpong {
             listener_liveness_max_sessions: 0,
             user_agent: "tari/pingpong/1.0.0".to_string(),
         };
+        let mut shutdown = Shutdown::new();
+        let shutdown_signal = shutdown.to_signal();
 
-        let (comms, dht) = rt
-            .block_on(initialize_comms(comms_config, publisher, vec![], Default::default()))
-            .unwrap();
-
-        println!("Comms listening on {}", comms.listening_address());
+        let transport_type = comms_config.transport_type.clone();
 
-        rt.block_on(comms.peer_manager().add_peer(peer_identity.to_peer()))
-            .unwrap();
-        let shutdown_signal = comms.shutdown_signal();
-
-        let handles = rt
-            .block_on(
-                StackBuilder::new(rt.handle().clone(), comms.shutdown_signal())
-                    .add_initializer(CommsOutboundServiceInitializer::new(dht.outbound_requester()))
-                    .add_initializer(LivenessInitializer::new(
-                        LivenessConfig {
-                            auto_ping_interval: None, // Some(Duration::from_secs(5)),
-                            ..Default::default()
-                        },
-                        Arc::clone(&subscription_factory),
-                        dht.dht_requester(),
-                    ))
-                    .finish(),
-            )
+        let mut handles = StackBuilder::new(shutdown.to_signal())
+            .add_initializer(P2pInitializer::new(comms_config, publisher, vec![]))
+            .add_initializer(LivenessInitializer::new(
+                LivenessConfig {
+                    auto_ping_interval: None, // Some(Duration::from_secs(5)),
+                    ..Default::default()
+                },
+                Arc::clone(&subscription_factory),
+            ))
+            .finish()
```

Comment:
nit: why not call this method `build()`

Reply:
No strong opinions - `finish` is widely used for builders and used consistently in comms. I don't mind changing it at all.